modules/tests/test_ref_modules/test_ref_modules.cpp

```diff
@@ -575,5 +575,493 @@ TEST_F(RefModulesTest, UpdateDevicePower)
+            ASSERT_EQ(static_cast<Float>(mark), expectedMark);
+            expectedMark += classCount;
+        }
+        ASSERT_EQ(labels[labels.getCount() - 1], classifierInputRange.getHighValue());
+    }
+
+    const auto classifierDomainSignal = classifierSignal.getDomainSignal();
+    const auto classifierDomainSignalDescription = classifierDomainSignal.getDescriptor();
+
+    const auto deviceDomainDescriptor = deviceSignal.getDomainSignal().getDescriptor();
+
+    const auto deviceDomainRule = deviceDomainDescriptor.getRule();
+    const auto classifierDomainRule = classifierDomainSignalDescription.getRule();
+
+    // Check Linear Rule
+    if (deviceDomainRule.getType() == DataRuleType::Linear)
+    {
+        ASSERT_EQ(deviceDomainRule.getType(), classifierDomainRule.getType());
+        ASSERT_EQ(classifierDomainSignalDescription.getTickResolution(), deviceDomainDescriptor.getTickResolution());
+
+        auto deviceResolution = deviceDomainDescriptor.getTickResolution() * Int(1000);
+
+        auto classifierDelta = classifierDomainRule.getParameters().get("delta");
+        auto expectedClassifierDelata = static_cast<Float>(blockSize) / static_cast<Float>(deviceResolution);
+
+        ASSERT_EQ(classifierDelta, expectedClassifierDelata);
+    }
+    else
+    {
+        ASSERT_EQ(classifierDomainRule.getType(), DataRuleType::Explicit);
+    }
+}
+
+TEST_F(RefModulesTest, ClassifierBlackBox)
+{
+    using inputSignalType = Int;
+    SampleType inputSignalSampleType = SampleType::Int64;
+    using outputSignalType = Float;
+
+    auto inputSignalRange = Range(0, 20);
+    bool inputSignalSync = true;
+
+    InputStubSignal input;
+    input.setUp(inputSignalSampleType, inputSignalRange, inputSignalSync);
+    auto instance = input.getInstance();
+    auto inputSignal = input.getSignal();
+
+    SizeT blockSizeMs = 5;
+    SizeT classCount = 1;
+    SizeT inputPacketCount = blockSizeMs;
+    SizeT outputPacketCount = inputPacketCount / blockSizeMs;
+
+    auto classifierFb = instance.addFunctionBlock("ref_fb_module_classifier");
+    classifierFb.setPropertyValue("UseCustomClasses", false);
+    classifierFb.setPropertyValue("BlockSize", blockSizeMs);
+    classifierFb.setPropertyValue("ClassCount", classCount);
+
+    const auto classifierPort = classifierFb.getInputPorts()[0];
+    classifierPort.connect(inputSignal);
+    const auto classifierSignal = classifierFb.getSignals()[0];
+
+    const auto classifierSignalDescription = classifierSignal.getDescriptor();
+    const auto classifierSignalLabelsSize = classifierSignal.getDescriptor().getDimensions()[0].getSize();
+
+    auto reader = BlockReader<outputSignalType>(classifierSignal, outputPacketCount);
+
+    auto dataPacket = input.createDataPacket(inputPacketCount);
+    auto dataPtr = static_cast<inputSignalType*>(dataPacket.getData());
+    for (size_t i = 0; i < inputPacketCount; i++)
+    {
+        auto deliter = inputPacketCount > 1 ? inputPacketCount - 1 : 1;
```

Comment:
The test cases are statically defined - they shouldn't need to have cases for checking what the inputs are, and what, for example, the `inputPacketCount` is. These are known at the start of the test.